hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/TokenTupleUtils.java

```diff
@@ -233,17 +236,30 @@ public static Tuple nftTokenInfoTupleFor(
             @NonNull final Nft nft,
             final long serialNumber,
             @NonNull final String ledgerId,
-            @NonNull final Account ownerAccount) {
-        requireNonNull(ownerAccount);
-        final var nftMetaData = nft.metadata() != null ? nft.metadata().toByteArray() : Bytes.EMPTY.toByteArray();
+            @NonNull final HederaNativeOperations nativeOperations) {
+        requireNonNull(nft);
+        requireNonNull(token);
+        requireNonNull(ledgerId);
+        requireNonNull(nativeOperations);
 
+        final var nftMetaData = nft.metadata() != null ? nft.metadata().toByteArray() : Bytes.EMPTY.toByteArray();
         return Tuple.of(
                 tokenInfoTupleFor(token, ledgerId),
                 serialNumber,
-                headlongAddressOf(ownerAccount),
+                // The odd construct allowing a token to not have a treasury account set is to accommodate
+                // Token.DEFAULT being passed into this method, which a few HtsCall implementations do
+                priorityAddressOf(nft.ownerIdOrElse(token.treasuryAccountIdOrElse(ZERO_ACCOUNT_ID)), nativeOperations),
                 nft.mintTimeOrElse(new Timestamp(0, 0)).seconds(),
                 nftMetaData,
-                headlongAddressOf(nft.spenderIdOrElse(ZERO_ACCOUNT_ID)));
+                priorityAddressOf(nft.spenderIdOrElse(ZERO_ACCOUNT_ID), nativeOperations));
+    }
+
+    private static Address priorityAddressOf(
+            @NonNull final AccountID accountId, @NonNull final HederaNativeOperations nativeOperations) {
+        requireNonNull(accountId);
+        return (ZERO_ACCOUNT_ID == accountId)
```

Comment:
Should we use `equals`?  I realize that `ZERO_ACCOUNT_ID` is static and probably what was used but what if someone creates a constant somewhere else?  Maybe I worry too much.

Reply:
This use of `ZERO_ACCOUNT_ID` is internal to `TokenTupleUtils`...could also make the first argument here `@Nullable` and then `return (accountId == null)`...semantics should be the same.

Will ponder doing elsewhere...but...passing checks 😳